ratpack-core/src/test/groovy/ratpack/http/client/HttpClientInterceptorSpec.groovy

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ratpack.http.client
+import spock.lang.Unroll
+
+/**
+ * Test suite for request/response interceptors for {@link HttpClient}
+ */
+class HttpClientInterceptorSpec extends HttpClientSpec {
+
+  @Unroll
+  def "can intercept #method request and response"() {
+    given:
+    otherApp {
+      get("foo") { ctx ->
+        render "bar"
+      }
+    }
+    def requestInterceptor = Mock(HttpClientRequestInterceptor)
+    def responseInterceptor = Mock(HttpClientResponseInterceptor)
+    when:
+
+    handlers {
+      get { ctx ->
+        ctx.execution.add(HttpClientRequestInterceptor, requestInterceptor)
+        ctx.execution.add(HttpClientResponseInterceptor, responseInterceptor)
```

Comment:
Here's where I wanted to use `bindings { ... }` to bind my mocks, but as far as I can tell, they're actually two different contexts(?)